OCPBUGS-36495: Add runbook for ClusterMonitoringOperatorDeprecatedConfig (#195)
openshift/runbooks: alerts/cluster-monitoring-operator/ClusterMonitoringOperatorDeprecatedConfig.md

# ClusterMonitoringOperatorDeprecatedConfig

## Meaning

This alert fires when a deprecated config is used in the config map `openshift-monitoring/cluster-monitoring-config`.

## Impact

Using a deprecated config is not recommended because the config has no effect and

> it may cause `cluster-monitoring-operator` go in degraded state when the config

Reviewer: Before we remove the field, we'll block the upgrades using the Upgradeable condition.

```suggestion
it may cause the Cluster Monitoring operator `Upgradeable` condition to become `False`  in a future OCP release.
```

> is removed in a future release.

Reviewer 2: ```suggestion
is no longer supported in a future release.
```

## Diagnosis

> - Check the __Description__ and __Summary__ sections of the alert to identify

Reviewer: ```suggestion
- Check the __Description__ and __Summary__ annotations of the alert to identify
```

the
  deprecated config as in the following example:

  __Description__

  `The configuration field k8sPrometheusAdapter in
  openshift-monitoring/cluster-monitoring-config was deprecated in 4.16 and has

>   no effect, you can remove it.`

Reviewer 2: I think we should remove `you can remove it` from the alert description.
Now we have a runbook + in some cases we also need to migrate the config...

Reviewer: Ideally the user shouldn't even have to read the runbook so I'd update the alert description to say the same about migrating the config.

  __Summary__

  `Cluster Monitoring Operator is being used with deprecated configuration.`

## Mitigation

> 

Reviewer 2: I think it's better if we handle each situation individually:
- For the `k8sPrometheusAdapter.dedicatedServiceMonitors` field, you can remove the block, see the release notes (entry) at XXX (for clusters <4.16)
- For the `k8sPrometheusAdapter` field, check the release notes entry at YYY, some of the subfields may need to be migrated under `metricsServer` (or sth like this)
